### PR TITLE
Manipulate list features

### PR DESCRIPTION
**Pull request by M0rrighan**
### Followed requirements: 

- Remove all hardcoded items from the tasks array.
- Create a new JavaScript file for the new functionality.
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for editing task descriptions.
- Implement a function for deleting a task (remove an element from the array).
- Implement a function for the "Clear all completed" button (use filter() method).
- By default new tasks should have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
- Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.
- All changes to the To Do List should be saved in local storage.

### CODE REVIEW

File: src/list.js [hunk 1]
```diff
@@ -28,4 +24,36 @@ export default class ListOfTasks {
...
+  removeByIndex(ind) {
+    const filtered = this.tasksList.filter((task) => task.index !== ind);
+    Storage.saveAndUpdate(this.overrideIndexes(filtered));
+    // Interact.populateUlTasksList(new ListOfTasks());
+  }
+
+  editDescription(index, newDescription) {
+    this.tasksList[index].description = newDescription;
+    return this.tasksList;
+  }
```
Comment (helman101) [inline src/list.js]: - [ ] You have done a wonderful job with the new functionality, but as the project requirements say, you should put this on a new Javascript file 🤓 
    - Kindly move this to another Javascript file, and you will be good to go 😄
    image

File: src/list.js [hunk 2]
```diff
@@ -28,4 +24,36 @@ export default class ListOfTasks {
...
+    const filtered = this.tasksList.filter((task) => task.done === false);
+    Storage.saveAndUpdate(this.overrideIndexes(filtered));
+    // Interact.populateUlTasksList(new ListOfTasks());
+  }
+
+  removeByIndex(ind) {
+    const filtered = this.tasksList.filter((task) => task.index !== ind);
+    Storage.saveAndUpdate(this.overrideIndexes(filtered));
+    // Interact.populateUlTasksList(new ListOfTasks());
+  }
```
Comment (helman101) [inline src/list.js]: - [ ] [OPTIONAL] There is an issue when removing a single element, whenever I have 2 tasks and I select the last one and then the first one and delete the first one, both tasks disappear. 🤨 
    - Here is a gif to show you exactly what happens. 🚀 
    issue

**Review by helman101 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 🔴 

Hi @M0rrighan  👋, 

You have done an excellent job working on this project 👏 👏👏
but there are still some requirements which aren't been addressed yet.

## To Highlight 🌟 
- You had done a great job passing all the linters.
- You had done a good job with your README file. 
- You had implemented the required new functions successfully. 🤩 

 ## Required Changes ♻️
We are almost there, but in order to complete this part and go ahead with the next project, you have to fix a couple of issues:

- [ ] Please check the comments below 🚀 

## Optional Changes 🤓 
Take a look at these [OPTIONAL] changes to improve your project:

- [ ] please check the comments below for [OPTIONAL] changes

Cheers and Happy coding!👏👏👏

fire

If anything is not 100% clear please leave your questions or comments in the PR thread, we will be glad to help you 😄.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by carlos-ssh — APPROVED** [review]
Hi @M0rrighan , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**